Mathlib/Tactic/Linarith/SimplexAlgo/SimplexAlgo.lean

```diff
@@ -0,0 +1,128 @@
+/-
+Copyright (c) 2024 Vasily Nesterov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vasily Nesterov
+-/
+import Mathlib.Tactic.Linarith.SimplexAlgo.Datatypes
+
+/-!
+# Simplex Algorithm
+
+To obtain required vector in `Linarith.SimplexAlgo.findPositiveVector` we run the Simplex Algorithm.
+We use Bland's rule for pivoting, which guarantees that the algorithm terminates.
+-/
+
+namespace Linarith.SimplexAlgo
+
+/-- An exception in the `SimplexAlgoM` monad. -/
+inductive SimplexAlgoException
+| Unfeasible : SimplexAlgoException
```

Comment:
The naming convention is that only types get `UpperCamelCase`, their inhabitants are `lowerCamelCase`.
```suggestion
| unfeasible : SimplexAlgoException
```

Reply:
(though perhaps this should be `infeasible`, given [these statistics](https://books.google.com/ngrams/graph?content=infeasible%2C+unfeasible&year_start=1800&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t1%3B%2Cinfeasible%3B%2Cc0%3B.t1%3B%2Cunfeasible%3B%2Cc0))